### PR TITLE
EmulatorPkg/Win/Host: Source level debugging on already loaded DLL

### PR DESCRIPTION
**Pull request by AshrafAliS**
# Description

Source level debugging is not happening for already loaded DLL file after temporary memory migration.
especially on PeiCore, PcdPeim and DxeCore.

- [ ] Breaking change?
  - **Breaking change** - Does this PR cause a break in build or boot behavior?
  - Examples: Does it add a new library class or move a module to a different repo.
- [ ] Impacts security?
  - **Security** - Does this PR have a direct security impact?
  - Examples: Crypto algorithm change or buffer overflow fix.
- [ ] Includes tests?
  - **Tests** - Does this PR include any explicit test code?
  - Examples: Unit tests or integration tests.

## How This Was Tested

 - Pick any PEI module loaded after temporary memory migration.
 - Before retuning from Entry Point add the breakpoint in Visual Studio.
 - Step in instruction, it will go AutoGen _ModuleEntryPoint, again step-in so that it will go to PeiCore
 - there Symbols will not be loaded from PeiCore (Error Message from Visual Studio as "The current stack frame was not found in a
 loaded module. Source cannot be shown for this location.")
 - After adding this patch, we can able to see C code in the Visual Studio Debugger from PeiCore.

## Integration Instructions

N/A

### CODE REVIEW

**Comment by AshrafAliS** [comment]
Before this patch:
image

After this patch:
image

**Comment by mdkinney** [comment]
There are other use cases for loading the same module more than once.  One is running an app more than once from the UEFI Shell or loading the same driver more than once.  This changes the behavior from the first image loading supporting source level debug to the last image loaded supporting source level debug.  Not sure if this change in policy is correct for all use cases.

File: EmulatorPkg/Win/Host/WinHost.c [hunk 1]
```diff
@@ -1131,15 +1131,9 @@ PeCoffLoaderRelocateImageExtraAction (
...
+      if ((Status == EFI_SUCCESS) || (Status == EFI_ALREADY_STARTED)) {
         //
-        // If the DLL has already been loaded before, then this instance of the DLL can not be debugged.
-        //
-        ImageContext->PdbPointer = NULL;
-        SecPrint ("WARNING: DLL already loaded.  No source level debug %S.\n\r", DllFileName);
-      } else {
-        //
-        // This DLL is not already loaded, so source level debugging is supported.
+        // This DLL is either not loaded or already started, so source level debugging is supported.
```
Comment (niruiyu) [inline EmulatorPkg/Win/Host/WinHost.c:1136]: Old logic was there for a reason.
I am not sure if it's safe to just use the DllEntryPoint for a loaded module.

Comment (AshrafAliS) [reply EmulatorPkg/Win/Host/WinHost.c:1136]: i checked the history of this code, and it introduced in the below check-in 
https://github.com/tianocore/edk2/commit/10ccc27c9548ab2550689e68f6f45dc48120bc79

Based on the description 
when the Driver is unloaded then it should remove the handle.

that we aren't changing from this PR.

may be i might i missed the complete picture of the PR where it introduced, can you check that PR once please

**Comment by AshrafAliS** [comment]
@niruiyu @ajfish help to review

**Comment by AshrafAliS** [comment]
@niruiyu @ajfish help to review

**Comment by AshrafAliS** [comment]
remainder

**Comment by AshrafAliS** [comment]
@ajfish @niruiyu @mdkinney any updates on the PR

**Review by niruiyu — APPROVED** [review]

**Comment by AshrafAliS** [comment]
Can we submit this patch

**Pull request merged**